File: app/src/main/java/eu/warble/pjapp/util/CredentialsManager.java
package eu.warble.pjapp.util;


import android.content.Context;

import eu.warble.pjapp.data.local.PreferencesManager;


public class CredentialsManager {

    private Credentials credentials;

    private CredentialsManager(){}

    private static class SingletonHelper{
        private static final CredentialsManager INSTANCE = new CredentialsManager();
    }

    public static CredentialsManager getInstance() {
        return SingletonHelper.INSTANCE;
    }

    public void saveCredentials(String login, String password, Context context){
        this.credentials = new Credentials(login, password);
        PreferencesManager.saveCredentials(login, password, context);
    }

    public void deleteCredentials(Context context){
        this.credentials = null;
        PreferencesManager.saveCredentials(null, null, context);
    }

    public Credentials getCredentials(Context context){
        if (credentials == null){
            String login = PreferencesManager.getLogin(context);
            String password = PreferencesManager.getPassword(context);
            if (login == null || password == null)
                return null;
            credentials = new Credentials(login, password);
        }
        return credentials;
    }

    public static class Credentials{
        public String login;
        public String password;

        public Credentials(String login, String password){
            this.login = login;
            this.password = password;
        }
    }
}

File: app/src/main/java/eu/warble/pjappkotlin/utils/CredentialsManager.kt
package eu.warble.pjappkotlin.utils

import android.app.Activity
import android.content.Context

object CredentialsManager {

    private val PJAPP_PREFS = "Pjapp-prefs"
    private var credentials: Credentials? = null

    fun saveCredentials(
            login: String?,
            password: String?,
            context: Context
    ) {
        this.credentials = Credentials(login, password)
        val preferences = context.getSharedPreferences(PJAPP_PREFS, Activity.MODE_PRIVATE)
        preferences.edit()
                .putString("login", login)
                .putString("password", password)
                .apply()
    }

    fun getCredentials(context: Context): Credentials? {
        if (credentials == null) {
            val preferences = context.getSharedPreferences(PJAPP_PREFS, Activity.MODE_PRIVATE)
            val login = preferences.getString("login", null)
            val password = preferences.getString("password", null)
            if (login == null || password == null) {
                return null
            }
            credentials = Credentials(login, password)
        }
        return credentials
    }

    data class Credentials(
            var login: String?,
            var password: String?
    )
}